docs/design/fabric-v2+/fpc-key-dist.puml

```diff
@@ -25,184 +24,153 @@ participant "TLCC_Enclave Peer1_Org1" as TLCC_Enclave1 #99FF99
     want to ensure that the private chaincode transaction
     can proceed in the case the running enclave is
     unavailable - for any possible or malicious cause.
-    Key confirmation can ensure that backup enclave are
+    Key registration can ensure that backup enclave are
     ready to take over and/or contribute.
   end note
-  group key-confirmation endorsement
-    Admin1  -> Admin1       :   key_confirmation_message <- <ECC1_S_PK, SIG_ECC1, CC_E_PK>
-    Admin1  -> ERCC1        ++: ConfirmCCKeys(key_confirmation_message)
-    ERCC1   -> ERCC1        :   check that creator is Admin
+  group chaincode-key-registration endorsement
+    Admin1  -> Admin1       :   cckey_registration_message <- <Enclave1_VK, Sig_Enclave1, Chaincode_EK>
+    Admin1  -> ERCC1        ++: RegisterCCKeys(cckey_registration_message)
+    ERCC1   -> ERCC1        :   check that tx proposal creator is Admin
     ERCC1   -> ERCC1        :   check that admin.org and enclave.org match
-    ERCC1   -> ERCC1        :   check ECC1_S_PK is registered
-    ERCC1   -> ERCC1        :   verify SIG_ECC1 over <CC_E_PK> using ECC1_S_PK
-    ERCC1   -> TLCC_Enclave1++: can_endorse(ECC1_S_PK)
-        note left
-            A secure channel with TLCC
-            is not required here.
-        end note
+    note over ERCC1
+        Check enclave registration by searching for its Credentials.
+    end note
+    ERCC1   -> ERCC1        :   Credentials <- find(Enclave1_VK)
+    ERCC1   -> ERCC1        :   verify Sig_Enclave1 over Chaincode_EK using Enclave1_VK
+    ERCC1   -> TLCC_Enclave1++: can_endorse(Enclave1_VK)
+    note over ERCC1
+        - A secure channel with TLCC is not required here.
+          Only the endorsement policy in the "current" chaincode definition is necessary.
+        - Notice that ERCC might have a stale chaincode definition, so ask TLCC. 
+        - This can_endorse function is a nice-to-have feature.
+          It prevents sharing keys with enclave that cannot endorse.
+        - We have to investigate how to implement this can_endorse function.
+    end note
     return true
-    note left
+    note over ERCC1
         if false, abort
     end note
-    ERCC1   -> ERCC1    :   put_state(key=<CC_name, CC_ver, ECC1_S_PK, "keydist_confirmed">, value=key_confirmation_message)
-        note left
-            This is the proof (stored on the ledger)
-            that the enclave has the CC keys.
-            In a multi-org setting, the proof informs
-            of enclave availability.
-        end note
+    ERCC1   -> ERCC1         :  extract CC_Id from Credentials
+    ERCC1   -> ERCC1         :  Registered_Chaincode_EK <- get_state(CC_Id."Chaincode_EK")
+    ERCC1   -> ERCC1         :  if(!Registered_Chaincode_EK) put_state(CC_Id."Chaincode_EK", cckey_registration_message.Chaincode_EK)
+    ERCC1   -> ERCC1         :  else check Registered_Chaincode_EK and cckey_registration_message.Chaincode_EK match
```

Comment:
that's why it was called "key confirmation": the second enclave is not registering the key, it is confirming that it has the right key.

Reply:
Actually, it is _not_ called "key confirmation" but "key registration"?!   

But even for key confirmation, for clients we anyway need a query which extracts the registered chaincode encryption key, why is this not sufficient also or an an admin to confirm that his local view is consistent with ercc? Seems easier (practical and conceptual) just to use the query to get the confirmation instead of trying a registration?

In any case, i can live with the else clause, just not what i would have expected and not obvious why it would be there.